plugins/catalog-backend/src/ingestion/processors/AnnotateScmSlugEntityProcessor.ts

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright 2021 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { Entity, LocationSpec } from '@backstage/catalog-model';
+import parseGitUrl from 'git-url-parse';
+import { identity, merge, pickBy } from 'lodash';
+import { CatalogProcessor } from './types';
+
+const GITHUB_ACTIONS_ANNOTATION = 'github.com/project-slug';
+
+export class AnnotateScmSlugEntityProcessor implements CatalogProcessor {
+  async preProcessEntity(
+    entity: Entity,
+    location: LocationSpec,
+  ): Promise<Entity> {
+    if (entity.kind !== 'Component' || location.type !== 'url') {
+      return entity;
+    }
+
+    const gitUrl = parseGitUrl(location.target);
```

Comment:
I'm wondering here how we can support Github Enterprise. Maybe not a requirement of this ticket, but I'm wondering if we should also be including the hostname in the project slug too? Thoughts @freben @Rugvip. At least with a processor like this, it's super nice to migrate to a different format further down the line too without having to update it in every repo 👍

Reply:
I would probably vote for renaming it to something GitHub actions specific and to include the host in the future. We just build a similar annotation internally that is also GitHub related and we choose to include the host for the exact reason.